src/SpriteManager.cpp

```diff
@@ -0,0 +1,44 @@
+#include "SpriteManager.h"
+
+SpriteManager::SpriteManager() {
+   loadFiles();
+}
+
+SpriteManager::~SpriteManager() {
+    for (auto const& surface : _sprites) {
```

Comment:
while this works it is more likely that you meant to use `const auto& surface` instead